internal/filter/regexTitle_test.go

```diff
@@ -0,0 +1,120 @@
+package filter
+
+import (
+	"testing"
+
+	"github.com/inovex/CalendarSync/internal/models"
+	"github.com/stretchr/testify/assert"
+)
+
+// Some events should be filtered
+func TestRegexTitleFilter(t *testing.T) {
+	sourceEvents := []models.Event{
+		{
+			ICalUID:     "testId",
+			ID:          "testUid",
+			Title:       "test",
+			Description: "bar",
+		},
+		{
+			ICalUID:     "testId2",
+			ID:          "testUid2",
+			Title:       "Test 2",
+			Description: "bar",
+		},
+		{
+			ICalUID:     "testId3",
+			ID:          "testUid2",
+			Title:       "foo",
+			Description: "bar",
+		},
+	}
+
+	expectedSinkEvents := []models.Event{
+		{
+			ICalUID:     "testId2",
+			ID:          "testUid2",
+			Title:       "Test 2",
+			Description: "bar",
+		},
+		{
+			ICalUID:     "testId3",
+			ID:          "testUid2",
+			Title:       "foo",
+			Description: "bar",
+		},
+	}
+
+	filter := RegexTitle{
+		Regexp: ".*test",
+	}
+
+	filteredEvents := []models.Event{}
+
+	for _, event := range sourceEvents {
```

Comment:
Done, had to add a "MockInterface" of the Filter Interface to prevent having an import loop. Is there an easier way to achieve this? 

left the assert outside of the helper func, may want to use different asserts for different test cases in the future.

Reply:
I've moved the `FilterEvents` helper into a `filter_test` package.

The alternative would be to somehow break the `Filter` interface out of the `sync` package, but I don't see an easy way.

Reply:
> I've moved the FilterEvents helper into a filter_test package.

That's what I did first, dismissed it as I thought introducing another package is not that clean 😂 
Thanks!

> The alternative would be to somehow break the Filter interface out of the sync package, but I don't see an easy way.

Yup, thought about this as well. Maybe we have to work on the whole code structure again, maybe I'll do that on a snowy winter day. But for now, this looks good.